doc/read-the-docs-site/quick-start.rst

```diff
@@ -0,0 +1,127 @@
+
+Writing the On-Chain Validator
+-----------------------------------------------------------------
+
+Prerequisites
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+GHC and Cabal should be installed in order to follow this guide.
+Plutus Tx currently supports GHC v9.2.x and v9.6.x.
+Cabal v3.8+ is recommended.
+
+Additionally, some C libraries will need to be installed.
+Plutus Tx depends on `cardano-base <https://github.com/input-output-hk/cardano-base>`_, which in turn depends on a few cryptographic C libraries, including ``libblst``, ``libsecp256k1``, and ``libsodium``.
+Cabal is not designed to manage C dependencies, so you need to either install them yourself, or manage them using another tool (such as Nix).
+
+If you are not using Nix, follow the instructions in the cardano-base repository to install these libraries (or install them in your own way if you like).
+
+If you are, you can use plutus repository's dev shell: ::
+
+  nix develop github:input-output-hk/plutus
+
+The dev shell comes with not only the required C libraries, but also GHC and Cabal.
+
+Create a New Cabal Package
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, make a new directory ``plutus-quickstart``: ::
+
+  mkdir plutus-quickstart && cd $_
+
+Then, create a new Cabal package that builds a library, using default settings: ::
+
+  cabal init --lib --non-interactive
+
+Declare Plutus Tx Dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Next, declare the following packages as dependencies: ``plutus-core``, ``plutus-ledger-api``, ``plutus-tx``, and ``plutus-tx-plugin``.
+These packages aren't on Hackage (the default repository Cabal looks for packages from), but on the `CHaP <https://github.com/input-output-hk/cardano-haskell-packages>`_ repository.
+We need to tell Cabal to look for packages from CHaP (in addition to Hackage).
+To do so, make a ``cabal.project`` file, and follow the instructions in the CHaP repository to make Cabal aware of CHaP.
+Then, add the following to your ``cabal.project``: ::
+
+  packages:
+    ./plutus-quickstart.cabal
+
+Finally, add the aformentioned packages in the ``build-depends`` field in ``plutus-quickstart.cabal``: ::
+
+  build-depends:
+    , base
+    , plutus-core
+    , plutus-ledger-api
+    , plutus-tx
+    , plutus-tx-plugin
+
+At this point, you should be able to build your project: ``cabal build plutus-quickstart`` should succeed.
+
+Write the Validator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now we are ready to write the on-chain validator using Plutus Tx.
+We showed and explained an auction validator in :ref:`simple_example`, and we'll reuse the same validator here.
+Make a file named ``AuctionValidator.hs``, and copy the content over from `here <https://github.com/input-output-hk/plutus/blob/master/doc/read-the-docs-site/tutorials/AuctionValidator.hs>`_.
+Add the following flag at the top of the file: ::
+
+  {-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:dump-uplc #-}
```

Comment:
I think it would be better to show people how to do this in their Haskell code, i.e. make an executable that writes the file to disk. This is rather weird, doing it as a side-effect of compilation.

Reply:
Great point. `dump-uplc` doesn't even work, because it is a parameterized contract, and an `AuctionParams` needs to be provided!